runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9122,15 +9256,72 @@ TR::CompilationInfo::compilationEnd(J9VMThread * vmThread, TR::IlGeneratorMethod
 
                   if (safeToStore)
                      {
+                     const U_8 *metadataToStore = dataStart;
+                     const U_8 *codedataToStore = codeStart;
+                     int metadataToStoreSize = dataSize;
+                     int codedataToStoreSize = codeSize;
+#ifdef COMPRESS_AOT_DATA
+                     if (!comp->getOption(TR_DisableAOTBytesCompression))
+                        {
+                        void * uncompressedData = comp->trMemory()->allocateHeapMemory(codeSize+dataSize);
+                        void * compressedData = comp->trMemory()->allocateHeapMemory(codeSize+dataSize);
```

Comment:
Though unlikely, it is possible for `allocateHeapMemory` to return NULL, so you might want to add NULL checks here.

Reply:
I take it back; what'll happen is that we'll throw an exception. I think you need to guard these two calls to `allocateHeapMemory` with a try/catch.